backend/infrahub/git/utils.py

```diff
@@ -46,3 +53,118 @@ async def get_repositories_commit_per_branch(
             )
 
     return repositories
+
+
+def _collect_parameter_first_segments(params: Any) -> set[str]:
```

Comment:
Do we need an `Any` here looks like we expect `str | list | tuple | set | dict` and won't use anything else.